shotwell does not build

shotwell does not build any more:
```
valac --target-glib=2.30.0 -g --enable-checking --fatal-warnings --ccode --enable-deprecated \
	--vapidir=../ --pkg=shotwell-plugin-dev-1.0 --pkg=gobject-2.0 --pkg=glib-2.0 --pkg=gdk-3.0 --pkg=gtk+-3.0 --pkg=gee-0.8 --pkg=libxml-2.0 --pkg=libsoup-2.4  \
	 \
	--vapidir=../common --pkg shotwell-plugin-common \
	--vapidir=../../vapi \
	shotwell-transitions.vala FadeEffect.vala SlideEffect.vala CrumbleEffect.vala BlindsEffect.vala CircleEffect.vala CirclesEffect.vala SquaresEffect.vala StripesEffect.vala ChessEffect.vala ClockEffect.vala
valac --target-glib=2.30.0 -g --enable-checking --fatal-warnings --ccode --enable-deprecated \
	--vapidir=../ --pkg=shotwell-plugin-dev-1.0 --pkg=gobject-2.0 --pkg=glib-2.0 --pkg=gdk-3.0 --pkg=gtk+-3.0 --pkg=gee-0.8 --pkg=gee-0.8 --pkg=libxml-2.0 --pkg=libsoup-2.4 --pkg=json-glib-1.0 --pkg=webkit2gtk-4.0  \
	 \
	--vapi=shotwell-plugin-common.vapi --header=shotwell-plugin-common.h \
	--vapidir=../../vapi \
	Resources.vala RESTSupport.vala
error: Package `shotwell-plugin-common' not found in specified Vala API directories or GObject-Introspection GIR directories
Compilation failed: 1 error(s), 0 warning(s)
../Makefile.plugin.mk:45: die Regel für Ziel „.stamp“ scheiterte
make[2]: *** [.stamp] Fehler 1
make[2]: Verzeichnis „/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/shotwell-transitions“ wird verlassen
Makefile:14: die Regel für Ziel „shotwell-transitions“ scheiterte
make[1]: *** [shotwell-transitions] Fehler 2
make[1]: *** Auf noch nicht beendete Prozesse wird gewartet …
cc -c -pthread -I/usr/include/gee-0.8 -I/usr/include/json-glib-1.0 -I/usr/include/webkitgtk-4.0 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/mirclient -I/usr/include/mircommon -I/usr/include/mircookie -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng12 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng12 -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/webkitgtk-4.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -nostdlib -export-dynamic -O2 -g -pipe  -fPIC -D_VERSION='"0.23.2"' -DGETTEXT_PACKAGE='"shotwell"' -I../.. -I ../common Resources.c
cc -c -pthread -I/usr/include/gee-0.8 -I/usr/include/json-glib-1.0 -I/usr/include/webkitgtk-4.0 -I/usr/include/gtk-3.0 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/x86_64-linux-gnu/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/mirclient -I/usr/include/mircommon -I/usr/include/mircookie -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/freetype2 -I/usr/include/libpng12 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng12 -I/usr/include/libsoup-2.4 -I/usr/include/libxml2 -I/usr/include/webkitgtk-4.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -nostdlib -export-dynamic -O2 -g -pipe  -fPIC -D_VERSION='"0.23.2"' -DGETTEXT_PACKAGE='"shotwell"' -I../.. -I ../common RESTSupport.c
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_argument_compare’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:122:9: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
         return strcmp(arg1.key, arg2.key);
         ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_argument_sort’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:131:9: warning: assignment from incompatible pointer type [-Wincompatible-pointer-types]
         return sorted_args.to_array();
         ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_transaction_check_response’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:218:10: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
         if (message.response_body.data == null || message.response_body.data.length == 0)
          ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:218:11: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
         if (message.response_body.data == null || message.response_body.data.length == 0)
           ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_transaction_get_response’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:352:9: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
         return (string) message.response_body.data;
         ^
/usr/share/vala-0.30/vapi/glib-2.0.vapi: In function ‘string_chug’:
/usr/share/vala-0.30/vapi/glib-2.0.vapi:1109:13: warning: passing argument 1 of ‘g_strchug’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   result._chug ();
             ^
In file included from /usr/include/glib-2.0/glib.h:79:0,
                 from RESTSupport.c:10:
/usr/include/glib-2.0/glib/gstrfuncs.h:175:8: note: expected ‘gchar * {aka char *}’ but argument is of type ‘const gchar * {aka const char *}’
 gchar*                g_strchug        (gchar        *string);
        ^
/usr/share/vala-0.30/vapi/glib-2.0.vapi: In function ‘string_chomp’:
/usr/share/vala-0.30/vapi/glib-2.0.vapi:1101:14: warning: passing argument 1 of ‘g_strchomp’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
   result._chomp ();
              ^
In file included from /usr/include/glib-2.0/glib.h:79:0,
                 from RESTSupport.c:10:
/usr/include/glib-2.0/glib/gstrfuncs.h:178:8: note: expected ‘gchar * {aka char *}’ but argument is of type ‘const gchar * {aka const char *}’
 gchar*                g_strchomp       (gchar        *string);
        ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_google_publisher_web_authentication_pane_on_page_load’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:761:2: warning: ‘gdk_cursor_new’ is deprecated [-Wdeprecated-declarations]
             pane_widget.get_window().set_cursor(new Gdk.Cursor(Gdk.CursorType.LEFT_PTR));
  ^
In file included from /usr/include/gtk-3.0/gdk/gdk.h:34:0,
                 from ../../plugins/shotwell-plugin-dev-1.0.h:15,
                 from RESTSupport.c:15:
/usr/include/gtk-3.0/gdk/gdkcursor.h:228:12: note: declared here
 GdkCursor* gdk_cursor_new   (GdkCursorType    cursor_type);
            ^
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala: In function ‘publishing_rest_support_google_publisher_web_authentication_pane_on_load_started’:
/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common/RESTSupport.vala:779:2: warning: ‘gdk_cursor_new’ is deprecated [-Wdeprecated-declarations]
             pane_widget.get_window().set_cursor(new Gdk.Cursor(Gdk.CursorType.WATCH));
  ^
In file included from /usr/include/gtk-3.0/gdk/gdk.h:34:0,
                 from ../../plugins/shotwell-plugin-dev-1.0.h:15,
                 from RESTSupport.c:15:
/usr/include/gtk-3.0/gdk/gdkcursor.h:228:12: note: declared here
 GdkCursor* gdk_cursor_new   (GdkCursorType    cursor_type);
            ^
cc  -L/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/install/lib -L/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/install/usr/lib -L/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/install/lib/x86_64-linux-gnu -L/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/install/usr/lib/x86_64-linux-gnu -shared Resources.o RESTSupport.o -lgee-0.8 -lxml2 -ljson-glib-1.0 -lwebkit2gtk-4.0 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lsoup-2.4 -lgio-2.0 -lgobject-2.0 -ljavascriptcoregtk-4.0 -lglib-2.0  -o libshotwell-plugin-common.so
make[2]: Verzeichnis „/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins/common“ wird verlassen
make[1]: Verzeichnis „/home/daniel/dev/snappy/snappy-playpen/shotwell/parts/shotwell/build/plugins“ wird verlassen
Makefile:686: die Regel für Ziel „plugins“ scheiterte
make: *** [plugins] Fehler 2
make: *** Auf noch nicht beendete Prozesse wird gewartet …
Command '['/bin/sh', '/tmp/tmp9e4scpll', 'make', '-j4']' returned non-zero exit status 2
daniel@daydream:~/dev/snappy/snappy-playpen/shotwell$ 
```

--- a/shotwell/shotwell-launcher
+++ b/shotwell/shotwell-launcher
@@ -1,3 +1,7 @@
 #! /bin/bash
+if [ ! -z "$1" ]
+  then
+    pwd=$PWD/
+fi
 cd $SNAP
-desktop-launch shotwell "$@"
+desktop-launch shotwell "$pwd$@"

--- a/shotwell/shotwell-launcher
+++ b/shotwell/shotwell-launcher
@@ -1,0 +1,3 @@
+#! /bin/bash
+cd $SNAP
+desktop-launch shotwell "$@"

--- a/shotwell/snapcraft.yaml
+++ b/shotwell/snapcraft.yaml
@@ -10,14 +10,19 @@ description: |
 confinement: strict
 
 apps:
-  shotwell-git:
-    command: desktop-launch shotwell
+  shotwell:
+    command: shotwell-launcher "$@"
     plugs: [x11, unity7, home, network]
 
 parts:
+  shotwell-launcher:
+    source: .
+    plugin: dump
+    organize:
+      shotwell-launcher: bin/shotwell-launcher
   shotwell:
     source: git://git.gnome.org/shotwell.git
-    #source-tag: shotwell-0.23.5
+    source-tag: shotwell-0.23.5
     plugin: autotools
     build-packages:
       - desktop-file-utils

--- a/shotwell/snapcraft.yaml
+++ b/shotwell/snapcraft.yaml
@@ -1,5 +1,5 @@
 name: shotwell
-version: 0.23.2
+version: 0.23.5
 summary: Shotwell photo manager
 description: |
   Shotwell is a digital photo organizer designed for the GNOME desktop
@@ -11,13 +11,13 @@ confinement: strict
 
 apps:
   shotwell-git:
-    command: gtk-launch shotwell
-    plugs: [x11, unity7, home]
+    command: desktop-launch shotwell
+    plugs: [x11, unity7, home, network]
 
 parts:
   shotwell:
     source: git://git.gnome.org/shotwell.git
-    source-tag: shotwell-0.23.2
+    #source-tag: shotwell-0.23.5
     plugin: autotools
     build-packages:
       - desktop-file-utils
@@ -74,4 +74,4 @@ parts:
       - librsvg2-common
       - dbus-x11
       - dconf-tools
-    after: [gtkconf]
+    after: [desktop-gtk3]

